From 7c1de5d7f0eb41a77bf90019755ecce4c8ca74d1 Mon Sep 17 00:00:00 2001
From: Xu Guangxin <guangxin.xu@intel.com>
Date: Mon, 16 Mar 2020 23:07:22 -0400
Subject: [PATCH 35/37] lavc: hevc, find_ref_idx, exclude current frame from
 long term refs

suppose
a. You have 3 frames, 0, 1, 4096.
b. The ltMask is 0xfff and use_msb is 0.
c. The 0, 1 are lt refs for 4096.
d. you are decode frame 4096, and get the 0 frame.
Since 4096 & ltMask is 0 too, even you want get 0, find_ref_idx may give you 4096.
add_candidate_ref will report an error for this
---
 libavcodec/hevc_refs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavcodec/hevc_refs.c b/libavcodec/hevc_refs.c
index 5d7bfc6a81..32e9ac7f89 100644
--- a/libavcodec/hevc_refs.c
+++ b/libavcodec/hevc_refs.c
@@ -389,7 +389,7 @@ static HEVCFrame *find_ref_idx(HEVCContext *s, int poc, uint8_t use_msb)
         for (i = 0; i < FF_ARRAY_ELEMS(s->DPB); i++) {
             HEVCFrame *ref = &s->DPB[i];
             if (ref->frame->buf[0] && ref->sequence == s->seq_decode) {
-                if ((ref->poc & LtMask) == poc)
+                if ((ref->poc & LtMask) == poc && (ref->poc != s->poc))
                     return ref;
             }
         }
-- 
2.20.1

